src/main/java/com/minecolonies/coremod/entity/ai/mobs/util/BarbarianSpawnUtils.java

```diff
@@ -0,0 +1,212 @@
+package com.minecolonies.coremod.entity.ai.mobs.util;
+
+import com.minecolonies.api.util.CompatibilityUtils;
+import com.minecolonies.api.util.constant.Constants;
+import com.minecolonies.coremod.MineColonies;
+import com.minecolonies.coremod.colony.Colony;
+import com.minecolonies.coremod.entity.EntityCitizen;
+import com.minecolonies.coremod.entity.ai.mobs.barbarians.*;
+import com.minecolonies.coremod.items.ModItems;
+import net.minecraft.entity.EntityList;
+import net.minecraft.entity.SharedMonsterAttributes;
+import net.minecraft.entity.ai.EntityAINearestAttackableTarget;
+import net.minecraft.entity.ai.EntityAISwimming;
+import net.minecraft.entity.ai.EntityAIWatchClosest;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.init.Items;
+import net.minecraft.inventory.EntityEquipmentSlot;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.ChunkPos;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.world.World;
+import net.minecraftforge.common.ForgeChunkManager;
+
+import java.util.stream.IntStream;
+
+/**
+ * Utils used for Barbarian Spawning
+ */
+public class BarbarianSpawnUtils
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add a private constructor to hide the implicit public one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1118)